### ISSUE TITLE
Upgrade to 4.9.0 from 4.8.x breaks dashboard-indexer communication

### ISSUE THREAD

**Issue by captainwasabi** [wazuh/wazuh#25605]
Ubuntu 22.04

I was installing logwatch on this server.  I did not do the wazuh install.  When installing a package I always `apt update` then `apt dist-upgrade`.  This caused it to upgrade wazuh to 4.9.0 (I assume from 4.8.2 since we do upgrades once a week).  I didn't replace any config files as part of the upgrade.  When I try to access it from the webpage I just get that the dashboard is not ready yet.  shut everything down, started  it back up, same result.  Started watching syslog as I was starting stuff, noted issue where things changed in the ossec.conf file. backed up the old one, moved the .new one in.  Compared them side by side, line by line.  Restarting wazuh-manager seems ok.  indexer seems ok. restarting dashboard has this:

```
Waiting until all OpenSearch nodes are compatible with OpenSearch Dashboards before starting saved objects migrations...
```

then
```
Unable to retrieve version information from OpenSearch nodes.
```

then a bunch of:

```
[ResponseError]: Response Error
```

I'm out of ideas. another issue here mentioned this and has a "fix" maybe in 4.9.1? but when will that be out?

**Comment by tkolaski** [comment]
I have the same problem after updating to 4.9

**Comment by tkolaski** [comment]
in wazuh-cluster.log I had entries:

Authentication finally failed for kibanaserver from 127.0.0.1:51086

I found:

https://groups.google.com/g/wazuh/c/rdCF0MBR6oU

It helped

**Comment by iasdeoupxe** [comment]
Had the same problem after updating from 4.8.2 to 4.9.0. After resetting the password of all users:
- https://documentation.wazuh.com/current/user-manual/user-administration/password-management.html#changing-the-passwords-for-all-users
- https://documentation.wazuh.com/current/user-manual/user-administration/password-management.html#passwords-distributed

and restarting all services the problem was solved for me.

**Comment by gdiazlo** [comment]
The problem comes from the lack of an environment variable OSD_PATH_CONF in `/etc/default/wazuh-dashboard`

This tells wazuh-dashboard where is the `opensearch_dashboard.keystore` which stores the credentials to connect to wazuh-indexer. 

When upgrading from 4.8.x to 4.9.0, this variable is lost, and dashboard starts using the keystore present in `/usr/share/wazuh-dashboard/config/` instead of the one which came from 4.8.x in `/etc/wazuh-dashboard`.

If you already changed the password, beware that if you add the missing env var missing, you need to copy your new keystore into the `/etc/wazuh/dashboard`, for example:

```
# cp /etc/wazuh-dashboard/opensearch_dashboard.keystore /etc/wazuh-dashboard/opensearch_dashboard.keystore.back
# cp /usr/share/wazuh-dashboard/config/opensearch_dashboard.keystore /etc/wazuh-dashboard/
```

**Comment by captainwasabi** [comment]
This time after  waiting for migrations, it started migrating.  Thank you, I never saw a reference to /etc/default in anything I looked at, and even so would not have know to add that.  So Thank you very much!

**Comment by Phurion** [comment]
I had to run what is below to make it that error go away.
`/usr/share/wazuh-indexer/plugins/opensearch-security/tools/wazuh-passwords-tool.sh --change-all`

**Comment by iasdeoupxe** [comment]
Updating the `wazuh-dashboard` package to `4.9.0-2` and following these steps worked:

https://github.com/wazuh/wazuh/issues/25605#issuecomment-2333884627

@gdiazlo In the instructions there is a minor typo as the `opensearch_dashboard.keystore` file is actually called `opensearch_dashboards.keystore` (with an `s` in `dashboards`).

@Phurion No need to set new passwords, see link above

**Comment by EHRETic** [comment]
Hi everybody,

As a few of us, I was impacted too.

Thanks a lot @gdiazlo , this saved my day! 👌😊
(I also didn't understand why to change a password that was never changed...)

**Comment by Callan-Breare** [comment]
> `/usr/share/wazuh-dashboard/bin/opensearch-dashboard-keystore`

This worked, thank you so much
